src/crab/type_encoding.hpp

```
@@ -51,6 +57,114 @@ std::string typeset_to_string(const std::vector<TypeEncoding>& items);
+
+    /// Build a TypeSet from a list of TypeEncoding values.
+    static constexpr TypeSet of(const std::initializer_list<TypeEncoding> types) {
+        TypeSet result = empty();
+        for (const auto te : types) {
+            result |= singleton(te);
+        }
+        return result;
+    }
+
+    // Lattice operations delegated to SmallBitsetDomain
+    constexpr TypeSet operator|(const TypeSet o) const { return TypeSet{(dom_ | o.dom_).raw()}; }
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)